src/app/CommandSender.cpp

```diff
@@ -92,9 +92,8 @@ CHIP_ERROR CommandSender::OnMessageReceived(Messaging::ExchangeContext * apExcha
 exit:
     ChipLogFunctError(err);
 
-    // Close the exchange cleanly so that the ExchangeManager will send an ack for the message we just received.
-    // This needs to be done before the Reset() call, because Reset() aborts mpExchangeCtx if its not null.
-    mpExchangeCtx->Close();
+    // Null out mpExchangeCtx, so our Shutdown() call below won't try to abort
+    // it and fail to send an ack for the message we just received.
     mpExchangeCtx = nullptr;
```

Comment:
This doesn't seem less error prone than the previous behavior.   It just seems different.  What if we don't set mpExchangeCtx to NULL in the callee?  If being called by the `ExchangeDelegate::OnMessageReceived` interface means the exchange manager now owns the exchange context, why don't simply make that happen automatically?

I think this could be done by moving mpExchangeCtx to `ExchangeDelegate` as a public member, or moving it to `ExchangeDelegate` as protected member and then making `ExchangeDelegate` a friend of `ExchangeManager`.

Reply:
@msandstedt See https://github.com/project-chip/connectedhomeip/pull/8050#discussion_r665824665 for the general goal here; this is step 1.

> What if we don't set mpExchangeCtx to NULL in the callee? 

Then things are bad, yes, after this PR.  The goal here is to not have an `mpExchangeCtx` in the callee in this case at all, but it'll take a few steps.

> I think this could be done by moving mpExchangeCtx to ExchangeDelegate as a public member,

Note that CommandSender isn't even an ExchangeDelegate, so that would not work for this specific case, unfortunately.

Reply:
I suppose the big reason not to do what I'm suggesting is that this would restrict implementers of `ExchangeDelegate` to only hold a single exchange context.  Still, it really seems like the right solution here is going to be something where everything is automatic.

Can we change the relevant IM interfaces to take `& mpExchangeCtx` so they, or something down the line, sets it null?

Reply:
Discussed offline.  Makes sense.  This is just the first step to get where we want to go, and is an improvement.